pkg/executor/insert.go

```diff
@@ -489,9 +564,12 @@ func (e *InsertExec) doDupRowUpdate(ctx context.Context, handle kv.Handle, oldRo
 			return nil
 		}
 	}
-	_, err := updateRecord(ctx, e.Ctx(), handle, oldRow, newData, assignFlag, e.Table, true, e.memTracker, e.fkChecks, e.fkCascades, dupKeyMode, e.ignoreErr)
-	if err != nil {
-		return err
+
+	if _, err := updateRecord(
```

Comment:
Is it possible to move the logic of handling generated column into the `updateRecord`? It seems that the function `updateRecord` also handles the `ON UPDATE` columns (and as we specified all `ON UPDATE` column in this functioin, these logic are meaningless).

Another possible solution is to remove the codes related to `ON UPDATE` columns in `updateRecord` function. However, as the same logic will be used multiple times (in `INSERT .. ON DUPLICATE UPDATE` and a normal `UPDATE` statement), I prefer to write the codes related to generated column in `updateRecord` to avoid repeating the codes.